src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,120 @@
+                                                     symbols: ImmutableList.Create(m),
+                                                     rules: CompletionItemRules.Default,
+                                                     contextPosition: position,
+                                                     properties: ImmutableDictionary<string, string>.Empty.Add("IsConversion", "true"));
+                    context.AddItems(allExplicitConversions);
+                }
+            }
+        }
+
+        internal override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem item, TextSpan completionListSpan, char? commitKey, bool disallowAddingImports, CancellationToken cancellationToken)
+        {
+            if (item.Properties.TryGetValue("IsConversion", out var value) && value == "true")
+            {
+                var completionChange = await HandleConversionChangeAsync(document, item, completionListSpan, cancellationToken).ConfigureAwait(false);
+                if (completionChange is { })
+                {
+                    return completionChange;
+                }
+            }
+
+            return await base.GetChangeAsync(document, item, completionListSpan, commitKey, disallowAddingImports, cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task<CompletionChange> HandleConversionChangeAsync(Document document, CompletionItem item, TextSpan completionListSpan, CancellationToken cancellationToken)
+        {
+            var symbols = await SymbolCompletionItem.GetSymbolsAsync(item, document, cancellationToken).ConfigureAwait(false);
+            var position = SymbolCompletionItem.GetContextPosition(item);
+            var symbol = symbols.Single() as IMethodSymbol;
+            var convertToType = symbol.ReturnType;
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindTokenOnLeftOfPosition(position);
+            var syntax = token.Parent;
+            var startPosition = completionListSpan.Start;
+            var typeName = convertToType.ToMinimalDisplayString(await document.ReuseExistingSpeculativeModelAsync(startPosition, cancellationToken).ConfigureAwait(false), startPosition);
+            if (syntax is MemberAccessExpressionSyntax memberAccess)
+            {
+                // expr. -> ((type)expr).
+                var newNode =
+                    ParenthesizedExpression(
+                        CastExpression(IdentifierName(typeName), memberAccess.Expression.WithoutTrivia()));
+                var newNodeText = newNode.ToFullString();
+                var textChange = new TextChange(memberAccess.Expression.Span, newNodeText);
+                return CompletionChange.Create(textChange);
+            }
+
+            return null;
+        }
+    }
+}
```

Comment:
overall i like this approach.  LMK when you want a full review though.  THanks! :)